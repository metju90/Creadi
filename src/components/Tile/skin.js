import { css } from 'styled-components';

export const TileWrapper = ({ theme }) => {
  const randomNumber = Math.floor((Math.random() * 23));
  return css`
  		text-align: center;
  		cursor: pointer;
		background: ${theme.colorSet[randomNumber]};
		border-radius: 0 0 0 15px;
		background-image: linear-gradient(to right,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.2) 100%);
		width:250px;
		height: 130px;
`;
};

export const Title = () => css`
	font-weight: 500;
`;

export const Premium = () => css`
	
`;
export const DeleteWrapper = () => css`
 	height: 130px;;
 	width: 70px;
 	background: #d9534f;
	background-image: linear-gradient(to right,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.2) 100%);
	border-radius: 0 15px 0 0;
	transform: translateX(70px);
    opacity: 1;
    position: absolute;
    right: 0;
    cursor: pointer;
    display: flex;
 `;

export const Wrapper = () => css`
	position: relative;
	display: inline-block;
	overflow: hidden;
	margin: 15px;
`;
export const Slider = () => css`
	width:100%;
    transition: opacity 0.1s, transform 0.25s;
    display:flex;
`;
