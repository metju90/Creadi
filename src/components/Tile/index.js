import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import * as skin from './skin';
import { removeInsurance } from '../../actions';

const TileWrapper = styled.div`${skin.TileWrapper}`;
const Title = styled.h2`${skin.Title}`;
const Premium = styled.span`${skin.Premium}`;
const DeleteWrapper = styled.div`${skin.DeleteWrapper}`;
const Wrapper = styled.div`${skin.Wrapper}`;
const Slider = styled.div`${skin.Slider}`;

class Tile extends Component {

	state = {
		isClicked:false,
	}

	 componentDidCatch(error, info) {
	    // You can also log the error to an error reporting service
	    console.log(error, info);
	 }

	tileClickEvent = () => this.setState(prevState => ({ isClicked: !prevState.isClicked}))
	
	 deleteInsurance = insurance =>  {
	 	// This is a very quic and not fully functional solution.
	 	// Just to let you know I did not forget this part of the UX =)
	 	window.confirm("Hey, are you sure about this?");
		this.props.removeInsurance(insurance);
	} 

  	render() {
	  	const { title, premium, index } = this.props;
	  	const { isClicked } = this.state;
	    return (
	      <Wrapper  onClick={this.tileClickEvent} >
	        <Slider style={ isClicked ? cssTransition : {}}>
		        <TileWrapper isClicked={isClicked}  index={index} >
		          <Title>{title}</Title>
		          <Premium>{`CHF ${premium}`}</Premium>
		        </TileWrapper>
		        <DeleteWrapper onClick={() => this.deleteInsurance(title)} > <FontAwesome style={{margin: "auto"}} size="2x" name="trash" /> </DeleteWrapper>
	        </Slider>
	      </Wrapper>
	    );
	  }
}

const cssTransition = {
  transform: 'translateX(-70px)',
};

Tile.propTypes = {
  title: PropTypes.string.isRequired,
  premium: PropTypes.string.isRequired,
};

const mapDispatchToProps = {
	removeInsurance,
}
export default connect(null, mapDispatchToProps)(Tile);
