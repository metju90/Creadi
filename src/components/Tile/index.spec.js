import React from 'react';
import Tile from './index';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const initialState = {};

const mockStore = configureStore();
let wrapper;
let store;

describe(' Tile Component', () => {
  beforeEach(() => {
  // creates the store with any initial state or middleware needed
	  store = mockStore(initialState);
    wrapper = shallow(<Tile store={store} />);
	 });

  // ...tests here...
  test(' CSS class changes when hovered', () => {
    const component = renderer.create(<Tile premium="100" title="Testing" />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter();

    // re-rendering
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
