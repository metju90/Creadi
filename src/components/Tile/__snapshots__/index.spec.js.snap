// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Tile Component  CSS class changes when hovered 1`] = `null`;

exports[`Login Component  CSS class changes when hovered 1`] = `null`;
